### PR TITLE
35 bill types

### PR DESCRIPTION
**Pull request by rehabas**
-Create bill types component and add style to it.

Selection_184

### CODE REVIEW

File: client/src/assets/index.js [hunk 1]
```diff
@@ -1,8 +1,6 @@
-import { ReactComponent as WifiIcon } from './Group 14.svg';
-import { ReactComponent as WaterIcon } from './Group 13.svg';
-import { ReactComponent as ElectIcon } from './Group 7.svg';
-import { ReactComponent as MobileIcon } from './smartphone-iphone.svg';
-
-export { default as expandArrow } from './expandArrow.svg';
+import { ReactComponent as ElectIcon } from './typesIcons/electricity.svg';
+import { ReactComponent as WifiIcon } from './typesIcons/wifi.svg';
+import { ReactComponent as WaterIcon } from './typesIcons/water.svg';
+import { ReactComponent as MobileIcon } from './typesIcons/smartphone-iphone.svg';
```
Comment (Amoodaa) [inline client/src/assets/index.js:4]: Remove icon from component name so we import like `Icons.Wifi`

File: client/src/assets/index.js [hunk 2]
```diff
@@ -1,8 +1,6 @@
-import { ReactComponent as WifiIcon } from './Group 14.svg';
-import { ReactComponent as WaterIcon } from './Group 13.svg';
-import { ReactComponent as ElectIcon } from './Group 7.svg';
-import { ReactComponent as MobileIcon } from './smartphone-iphone.svg';
-
-export { default as expandArrow } from './expandArrow.svg';
+import { ReactComponent as ElectIcon } from './typesIcons/electricity.svg';
```
Comment (Amoodaa) [inline client/src/assets/index.js:1]:
```suggestion
export { ReactComponent as ElectIcon } from './typesIcons/electricity.svg';
```

File: client/src/pages/Home.js [hunk 3]
```diff
@@ -1,5 +1,49 @@
...
+        className="bg-magenta lg:w-1/2 md:w-2/3 border-l-8 border-kournikova text-mich-white p-4"
+        role="alert"
+      >
+        <p className="font-bold">Error</p>
+        <p>{error}</p>
+      </div>
+    );
+
+  if (!billTypes?.length)
+    return <Spinner className="w-8 h-8 m-auto spin mt-64" />;
```
Comment (Amoodaa) [inline client/src/pages/Home.js:40]: Extract the spinner from `App.js` and reuse it here, we shouldnt have more than one spinner for loading

File: client/src/pages/Home.js [hunk 4]
```diff
@@ -37,7 +37,7 @@ const Home = () => {
     );
 
   if (!billTypes?.length)
-    return <Spinner className="w-8 h-8 m-auto spin mt-64" />;
+    return <Loader className="w-8 h-8 m-auto spin mt-64" />;
```
Comment (Amoodaa) [inline client/src/pages/Home.js:40]: No need for classes

File: client/src/pages/Home.js [hunk 5]
```diff
@@ -1,5 +1,50 @@
...
+      const res = await axios.get('/api/v1/bills/me');
+      const { data: bills } = res;
+      const allTypes = [];
+      const tempTypes = {};
+
+      bills.forEach(({ type: { name }, type_id: id }) => {
+        if (!tempTypes[name]) allTypes.push({ name, id });
+        tempTypes[name] = 'exists';
+      }, {});
+      allTypes.sort((a, b) => (a.id > b.id ? 1 : -1));
```
Comment (AhmedSafi97) [inline client/src/pages/Home.js:20]: You can extract this logic as a function!

Comment (AhmedSafi97) [reply client/src/pages/Home.js:20]: don't  forget to test it :stuck_out_tongue:

File: client/src/pages/Home.js [hunk 6]
```diff
@@ -6,27 +6,29 @@ const Home = () => {
...
-      setUserBillTypes(allTypes);
-    } catch (err) {
-      if (err.response) setError(err.response.data.message);
-      else setError('Something went error');
-    }
+    bills.forEach(({ type: { name }, type_id: id }) => {
+      if (!tempTypes[name]) allTypes.push({ name, id });
+      tempTypes[name] = 'exists';
+    }, {});
+    return allTypes.sort((a, b) => (a.id > b.id ? 1 : -1));
```
Comment (Amoodaa) [inline client/src/pages/Home.js:17]:
```suggestion
    return allTypes.sort((a, b) => (a.id - b.id));
```

**Pull request merged**